packages/fieldset/src/LionFieldset.js

```diff
@@ -84,28 +93,46 @@ export class LionFieldset extends FormRegistrarMixin(
     super();
     this.disabled = false;
     this.submitted = false;
+    this.dirty = false;
+    this.touched = false;
+    this.focused = false;
     this.formElements = {};
     this.__addedSubValidators = false;
     this.__createTypeAbsenceValidators();
+
+    this._checkForOutsideClick = this._checkForOutsideClick.bind(this);
   }
 
   connectedCallback() {
     // eslint-disable-next-line wc/guard-super-call
     super.connectedCallback();
+
+    this.addEventListener('focusin', this._updateTouchedClass);
+    this.addEventListener('focusout', this._onFocusOut);
+    this.addEventListener('focusin', this._syncFocused);
+
     this.addEventListener('validation-done', this.__validate);
-    this.addEventListener('focused-changed', this._updateFocusedClass);
-    this.addEventListener('touched-changed', this._updateTouchedClass);
-    this.addEventListener('dirty-changed', this._updateDirtyClass);
+    this.addEventListener('dirty-changed', this._syncDirty);
+
     this._setRole();
+    document.addEventListener('click', this._checkForOutsideClick);
+  }
+
+  _checkForOutsideClick(event) {
```

Comment:
what if you have 2 fieldsets, then both will be touched?

Reply:
done